src/ts/contentScripts/entry/index.ts

```diff
@@ -5,4 +5,53 @@
 
 import '../../features/features'
 import '../../features/dispatcher'
-import '../../features/fuzzy_date'
\ No newline at end of file
+import '../../features/fuzzy_date'
+
+//detect when roam is loaded
+const target = document.getElementById('app')
+if (target) {
+    // Wait for Roam to load before initializing
+    const observer = new MutationObserver(function (mutations) {
+        mutations.forEach(function (mutation) {
+            const newNodes = mutation.addedNodes // DOM NodeList
+            if (newNodes !== null) {
+                // If there are new nodes added
+                for (let node of newNodes as NodeListOf<Element>) {
+                    if (node.classList.contains('roam-body')) {
+                        // body has loaded
+                        console.log('roam loaded')
+                        document.dispatchEvent(new CustomEvent('roam-loaded', {detail: new Date()}))
+                        observer.disconnect()
+
+                        //for the first 60 seconds, run the scripts every 3 seconds to load initially
+                        for (let time = 0; time <= 60000; time += 3000) {
+                            setTimeout(function () {
+                                document.dispatchEvent(new CustomEvent('roam-toolkit-tick', {}))
+                            }, time)
+                        }
+
+                        //after that, run the scripts every 15 seconds indefinately
+                        setInterval(function () {
+                            document.dispatchEvent(new CustomEvent('roam-toolkit-tick', {}))
```

Comment:
I'm not entirely sure why do we need to do this actually? If I understand MutationObserver correctly - we can observe changes in the individual blocks instead. Which seems like it'd provide a more responsive results vs periodically iterating over all the blocks?  
I've done a similar thing with Asana tasks a while ago: https://github.com/Stvad/Asana-counter/blob/master/Asana%20counter.user.js#L26

There we observe each existing task, and have a separate thing to enable observation on the new blocks as they are loaded. Not sure if that's the best approach (maybe we can just have observer for any kind of changes in the blocks?) but seems like doing something similar should allow us to do a more targeted rendering..

Reply:
Because I don't believe that we can use an observer to notice changes to all current and future changes, and because you need it to run these functions even if no changes are being made - the contents can change from other users or when you navigate to a new page but have not yet made changes.

Without being in control of the underlying code, and especially since we expects this code to change, this is by far the safest option.